src/boot/grub2/ostree-grub-generator

```diff
@@ -0,0 +1,104 @@
+#!/bin/sh
+
+# To use a custrom script for generating grub.cfg, set the --with-grub2-mkconfig=no
+# configure switch when configuring and building OSTree.
+#
+# This script is called by ostree/src/libostree/ostree-bootloader-grub2.c whenever
+# boot loader configuration file needs to be updated. It can be used as a template
+# for a custom grub.cfg generator. What to consider when writing a custom grub.cfg
+# generator:
+#
+#   - The populate_menu() function converts boot loader entries as defined by
+#   https://www.freedesktop.org/wiki/Specifications/BootLoaderSpec/ into GRUB2
+#   menuentry sections. This is the core logic that is required by OSTree
+#   based system.
+#
+#   - Embedded systems: Be aware that this script is executed not only on a host machine by OS
+#   installer, but also on a target device, thus think about shell portability. A target device
+#   for example might be using busybox with a limited shell.
+#
+# Feel free to edit this script to fit your requirements.
+
+set -e
+
+script=$(basename ${0})
+# Atomically safe location where to generete grub.cfg when executing system upgrade.
+new_grub2_cfg=${2}
+entries_path=$(dirname $new_grub2_cfg)/entries
+
+read_config()
+{
+    config_file=${entries_path}/${1}
+    title=""
+    initrd=""
+    options=""
+    linux=""
+
+    while read -r line
+    do
+        record=$(echo ${line} | cut -f 1 -d ' ')
+        value=$(echo ${line} | cut -s -f2- -d ' ')
+        case "${record}" in
+            "title")
+                title=${value}
+                ;;
+            "initrd")
+                initrd=${value}
+                ;;
+            "linux")
+                linux=${value}
+                ;;
+            "options")
+                options=${value}
+                ;;
+        esac
+    done < ${config_file}
+
+    if [ -z "${title}" ]; then
+        title="(Untitled)"
+    fi
+}
+
+populate_menu()
+{
+    # Prepend "boot/" prefix when boot/ directory is on the same partition as usr/,
+    # if boot/ directory is on a separate partition, set boot_prefix=""
+    boot_prefix="/boot"
```

Comment:
See https://github.com/ostreedev/ostree/pull/215 - I wonder if we could have OSTree pass this down via environment variable?  Though that only works on-target.  Perhaps anyone doing a cross system like OpenEmbedded could have that set the `OSTREE_BOOT_PARTITION` environment variable there?